cipher-asymmetric/src/main/scala/tsec/cipher/asymmetric/core/AsymmetricCipherAlgebra.scala

```diff
@@ -0,0 +1,15 @@
+package tsec.cipher.asymmetric.core
+
+import tsec.cipher.asymmetric.imports._
+import tsec.cipher.common.mode.NoMode
+import tsec.cipher.common.{CipherText, PlainText}
+
+trait AsymmetricCipherAlgebra[F[_], A, P] {
+  type C
```

Comment:
Why internal typing here over a typed announced into the trait? As it appears only to be used in a location that would easily lend to this.

`trait AsymmetricCipherAlgebra[F[_], A, P, C]`

`extends AsymmetricCipherAlgebra[Either[CipherError, ?], A, P, JCipher]` 

However I'm still exploring the code base, so perhaps I just dont' understand yet.

Reply:
@ChristopherDavenport this is more my design decision, as I wanted to maintain the cipher's representation as an internal thing. This algebra is very similar to [this](https://github.com/jmcardon/tsec/blob/c3566c48027035b2b7b0a1346e77e64a721a3e85/cipher-symmetric/src/main/scala/tsec/cipher/symmetric/core/SymmetricCipherAlgebra.scala).

In particular, it's mostly because I wanted to not expose the actual representation of the internal state of the cipher in the type signature, but treat it as sort of an "internal state type" that wouldn't usually escape to the outside world, unless you absolutely want it for whatever reason.

Reply:
The logic behind it, mostly, is that if we ever decide to be polymorphic about the cipherAlgebra we're using, i.e we want to be polymorphic over `F[_], A, M, P`, I don't want a consumer (or myself) to have to pass the internal representation type to something that consumes it.